main.tf

```diff
@@ -1,96 +1,109 @@
 provider "aws" {
-  access_key = "${var.access_key}"
-  secret_key = "${var.secret_key}"
-  region = "${var.region}"
+  region = var.region
 }
 
-# VPC creation
-resource "aws_vpc" "minecraft_vpc" {
-  cidr_block = "${var.vpc_cidr}"
-  enable_dns_support = true
-  enable_dns_hostnames = true
-}
+module "vpc" {
```

Comment:
I've been thinking about this specific module choice for a while. What I don't like about it is it will create a lot of unnecessary subnets and NAT gateway which isn't needed. Virtually every other improvement I do like.

Reply:
> I've been thinking about this specific module choice for a while. What I don't like about it is it will create a lot of unnecessary subnets and NAT gateway which isn't needed. Virtually every other improvement I do like.

NAT Gateways are necessary for egress traffic out to the internet. If there were no NATs setup, no instance would have access to the internet: https://github.com/terraform-aws-modules/terraform-aws-vpc#nat-gateway-scenarios. See [private vs. intra subnets](https://github.com/terraform-aws-modules/terraform-aws-vpc#private-versus-intra-subnets) section for more on that.

Subnets are also necessary as well, otherwise you wouldn't be able to create instances. You could specify only 1 range and end up with 1 subnet, if you wanted.

Reply:
The intent for this specific project is one EC2 instance in a publicly accessible subnet. The NAT gateway would not be needed since the intent is the instance would be going in a public subnet. It would just need an IGW attached via route table and a proper SG for ingress and egress (https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Internet_Gateway.html). I think the private subnets can be totally excluded from this all together. The module in the PR has 2 public subnets and 2 private subnets with NAT gateways included. The NAT Gateways would only be used if the instances are created in the private subnet. (https://docs.aws.amazon.com/vpc/latest/userguide/vpc-nat.html). The version of this project using ECS with containers that will be created in a private subnet works excellent with the module you used in the PR, since awsvpc networking doesn't allow instances to be in a public subnet if they aren't Fargate, so they would need to be in a private subnet with a NAT Gateway.

The master branch has the one public subnet, with an IGW attached and while it does take a bit more code, it may be a better solution that using the full VPC module because you wont have the unused subnets and NAT Gateways, just the one IGW and public subnet.

Reply:
Ah, good point. I believe you can still use this module to have 0 private subnets. Just pass in an empty list.